jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/JavaContainerBuilderHelper.java

```diff
@@ -46,22 +52,43 @@
    */
   public static FileEntriesLayer extraDirectoryLayerConfiguration(
       Path extraDirectory,
-      Map<AbsoluteUnixPath, FilePermissions> extraDirectoryPermissions,
+      Map<String, FilePermissions> extraDirectoryPermissions,
       BiFunction<Path, AbsoluteUnixPath, Instant> modificationTimeProvider)
       throws IOException {
     FileEntriesLayer.Builder builder =
         FileEntriesLayer.builder().setName(LayerType.EXTRA_FILES.getName());
+    Map<String, PathMatcher> pathMatchers = new HashMap<>();
```

Comment:
Hmm... Isn't `Map<PathMatcher, Permissions>` more natural? Inside `walk()`, you iterate over this map instead of `extraDirectoryPermissions`.

Reply:
Oh, yeah. Wasn't seeing it at the time but I get it now.